core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -796,14 +797,18 @@ private void forwardSetModeRequestToMaster(
   }
 
   private void canonicalizeSchema(Schema schema) throws InvalidSchemaException {
-    if (schema == null || schema.getSchema().trim().isEmpty()) {
+    if (schema == null
+        || schema.getSchema() == null
+        || schema.getSchema().trim().isEmpty()) {
+      log.error("Empty schema");
       throw new InvalidSchemaException("Empty schema");
```

Comment:
Here and below: we already throw, why the log message ?

Reply:
Just to make sure it appears in the log.